server/src/test/java/io/druid/metadata/SQLMetadataRuleManagerTest.java

```diff
@@ -109,11 +109,48 @@ public void testAuditEntryCreated() throws Exception
     List<AuditEntry> auditEntries = auditManager.fetchAuditHistory("test_dataSource", "rules", null);
     Assert.assertEquals(1, auditEntries.size());
     AuditEntry entry = auditEntries.get(0);
-    Assert.assertEquals(mapper.writeValueAsString(rules),entry.getPayload());
-    Assert.assertEquals(auditInfo,entry.getAuditInfo());
+    Assert.assertEquals(mapper.writeValueAsString(rules), entry.getPayload());
+    Assert.assertEquals(auditInfo, entry.getAuditInfo());
     Assert.assertEquals("test_dataSource", entry.getKey());
   }
 
+  @Test
+  public void testFetchAuditEntriesForAllDataSources() throws Exception
+  {
+    List<Rule> rules = Arrays.<Rule>asList(
+        new IntervalLoadRule(
+            new Interval("2015-01-01/2015-02-01"), ImmutableMap.<String, Integer>of(
+            DruidServer.DEFAULT_TIER,
+            DruidServer.DEFAULT_NUM_REPLICANTS
+        )
+        )
+    );
+    AuditInfo auditInfo = new AuditInfo("test_author", "test_comment", "127.0.0.1");
+    ruleManager.overrideRule(
+        "test_dataSource",
+        rules,
+        auditInfo
+    );
+    ruleManager.overrideRule(
+        "test_dataSource2",
+        rules,
+        auditInfo
+    );
+    // fetch rules from metadata storage
+    ruleManager.poll();
+
+    Assert.assertEquals(rules, ruleManager.getRules("test_dataSource"));
+    Assert.assertEquals(rules, ruleManager.getRules("test_dataSource2"));
+
+    // test fetch audit entries
+    List<AuditEntry> auditEntries = auditManager.fetchAuditHistory("rules", null);
+    Assert.assertEquals(2, auditEntries.size());
+    for (AuditEntry entry : auditEntries) {
+      Assert.assertEquals(mapper.writeValueAsString(rules), entry.getPayload());
```

Comment:
is it possible to check object equality instead of string equality by using...
```
Assert.assertEquals(rules, mapper.readValue(entry.getPayload(), typeref..List);
```